backend/instancemgmt/locker_test.go

```diff
@@ -0,0 +1,48 @@
+package instancemgmt
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLocker(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Tests with Sleep")
+	}
+	const notUpdated = "not_updated"
+	const atThread1 = "at_thread_1"
+	const atThread2 = "at_thread_2"
+	t.Run("Should lock for same keys", func(t *testing.T) {
+		updated := notUpdated
+		locker := newLocker()
+		locker.Lock(1)
+		defer locker.Unlock(1)
+		go func() {
+			locker.RLock(1)
+			defer locker.RUnlock(1)
+			require.Equal(t, atThread1, updated, "Value should be updated in different thread")
+			updated = atThread2
+		}()
+		time.Sleep(time.Millisecond * 10)
+		require.Equal(t, notUpdated, updated, "Value should not be updated in different thread")
+		updated = atThread1
+	})
+
+	t.Run("Should not lock for different keys", func(t *testing.T) {
+		updated := notUpdated
+		locker := newLocker()
+		locker.Lock(1)
+		defer locker.Unlock(1)
+		go func() {
+			locker.RLock(2)
+			defer locker.RUnlock(2)
+			require.Equal(t, notUpdated, updated, "Value should not be updated in different thread")
+			updated = atThread2
+		}()
+		time.Sleep(time.Millisecond * 10)
```

Comment:
I agree that 10 ms sleep can\`t guarantee that another goroutine will start and set the value to variable, probably if test run on some really slow machine it can be flaky. Only usage of synchronization primitives could guarantee it, but here we are testing the work of synchronization primitives, so we can\`t use it. I used some small reasonable value for sleep, also I put this test under the `testing.Short()`. Not sure how we could avoid it here.